rfc-120-csv-preview.md

```diff
@@ -0,0 +1,134 @@
+  },
+  edition_links: {
+    parent: {
+      description: "The parent content item.",
+      maxItems: 1,
+    },
+  }
+}
+```
+
+- Using a content item is consistent with the one-to-one mapping of pages and content on GOV.UK. This should make it easier to debug issues with the feature, since it follows an existing pattern.
+
+- The schema includes a link to the parent document. Using an expanded link means we don't need to make an extra API call to the Content Store in order to show information about the parent document.
+
+- Using the publishing system means we can follow an existing approach for things like automatic redirects (in case the base path changes) and anonymous preview ('fact check'). However, this entails an overhead of extra API calls to manage the separate content items, with the risk they get out-of-sync.
+
+- Using a separate content item means we're not coupled to a particular rendering app. This provides a foundation for extending the preview feature in the future.
+
+### Step 2
+
+Re-implement the CSV Preview feature in Government Frontend.
+
+- Government Frontend currently renders all documents published by Whitehall, that have CSV file attachments. This makes it a relevant codebase to implement this feature in.
+
+- We expect the implementation to be small, isolated and not to grow/change much. Adding a new responsibility to Government Frontend seems like a reasonable compromise when compared to the [rejected proposals](#rejected-proposals).
+
+
+### Step 3
+
+Change Whitehall to use the new CSV Preview feture for newly published documents. We will need to use a different URL structure, since the current one is coupled to Whitehall.
+
+```
+# current URL example
+https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/856932/Management_information_-_further_education_and_skills_-_as_at_31_December_2019.csv/preview
+
+# proposed URL example
+https://www.gov.uk/government/remaining-parent-document-base-path/preview/asset-id
+```
+
+The `asset-id` should be the ID used by Asset Manager e.g. `597b098a759b743e0b759a96`, since it persists for the lifetime of the asset, even if the original file is replaced.
+
+
+### Step 4
+
+Change existing attachments (in Whitehall) to use the new CSV Preview. This will involve republishing all Whitehall documents that include a CSV preview link.
+
+Once this is done, we can remove some of the config for [the `assets` domain](https://github.com/alphagov/govuk-puppet/blob/51d7f7b648e8fde5aae3adfb774ec3bca6325bd8/modules/router/templates/assets_origin.conf.erb#L69-L74) that supported the legacy feature ([`gov.uk` routing](https://github.com/alphagov/government-frontend/blob/dcbc563030c470e6242d6432896d84feebb2138b/config/routes.rb#L14) remains). We can also delete the code for the legacy feature.
+
+> We may want to add redirects for the old preview URLs in Whitehall, if they continue to receive lots of traffic after all the 'View Online' links have updated.
```

Comment:
Might be worth adding a line saying we can use Google Analytics to help inform this decision. Or would it be CDN logs? 🤔

Reply:
I think we can leave that as a detail for the person doing it. Either should work, although CDN logs would be more reliable these days.